### PR TITLE
feat: upgrade `prettier` to v2

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Re-submission of https://github.com/mysticatea/eslint-plugin/pull/39

> This impacts the `@mysticatea/prettier` rule
> 
> I've also fixed all ESLint errors this caused & ran Prettier on the full codebase

---

BREAKING CHANGE: `es5`, `es2015`, `es2016`, `es2017`, `es2018`, `es2019`, `es2020`, `es2021` & `es2022` configs will now report for their changed default values. For more info check their [release notes](https://prettier.io/blog/2020/03/21/2.0.0.html#breaking-changes)

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@eslint-community/eslint-tsc @eslint-community/core-team @eslint-community/mysticatea-eslint-plugin Since [this plugin is adhering the same semantic versioning policy as ESLint](https://github.com/eslint-community/eslint-plugin-mysticatea#-semantic-versioning-policy), this PR would need to be released in a major release I guess?

File: docs/rules/arrow-parens.md [hunk 1]
```diff
@@ -12,39 +12,45 @@ This rule ignores arrow functions that there is a open paren before itself.
 ### The following patterns are considered warnings:
 
 ```js
-const twice = x => 2 * x;
+const twice = (x) => 2 * x
```
Comment (ota-meshi) [inline docs/rules/arrow-parens.md]: I think this file should probably not be formatted.

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: docs/rules/arrow-parens.md (hunk 1 above)
Comment (MichaelDeBoey) [reply docs/rules/arrow-parens.md]: This is because I ran `prettier` on all files, so I think it's OK

Comment (ota-meshi) [reply docs/rules/arrow-parens.md]: Formatting the example code in this file would not be a good explanation of the rule.
This should explain the violation of this rule related to parentheses.

Comment (MichaelDeBoey) [reply docs/rules/arrow-parens.md]: Oh your talking about this specific file, I thought you were talking about all doc files 🙈

Will revert 👍

File: docs/rules/arrow-parens.md [hunk 2]
```diff
@@ -12,39 +12,45 @@ This rule ignores arrow functions that there is a open paren before itself.
...
-xs.map(x => 2 * x);
+xs.map((x) => 2 * x)
 ```
 
 ```js
-p.then(x => 2 * x, (err) => console.error(err));
+p.then(
+    (x) => 2 * x,
+    (err) => console.error(err)
+)
```
Comment (ota-meshi) [inline docs/rules/arrow-parens.md]: We need to revert the line breaks.
The example probably describes the following test case:
https://github.com/eslint-community/eslint-plugin-mysticatea/blob/0f9086016c7c13cdaa7c6c9865834475c1642cb3/tests/lib/rules/arrow-parens.js#L23

```suggestion
p.then(x => 2 * x, (err) => console.error(err))
```

File: docs/rules/arrow-parens.md [hunk 3]
```diff
@@ -12,39 +12,45 @@ This rule ignores arrow functions that there is a open paren before itself.
...
 const obj2 = {
-  twich: (x => 2 * x)
-};
+    twich: (x) => 2 * x,
+}
 ```
 
 ```js
-xs.map(x => 2 * x);
+xs.map((x) => 2 * x)
```
Comment (ota-meshi) [inline docs/rules/arrow-parens.md]: We need to revert the parenthesis.
The example probably describes the following test case:
https://github.com/eslint-community/eslint-plugin-mysticatea/blob/0f9086016c7c13cdaa7c6c9865834475c1642cb3/tests/lib/rules/arrow-parens.js#L17

```suggestion
xs.map(x => 2 * x)
```

File: docs/rules/arrow-parens.md [hunk 4]
```diff
@@ -12,39 +12,45 @@ This rule ignores arrow functions that there is a open paren before itself.
...
 ```js
 const obj = {
-  twich: (x) => 2 * x
-};
+    twich: (x) => 2 * x,
+}
 const obj2 = {
-  twich: (x => 2 * x)
-};
+    twich: (x) => 2 * x,
```
Comment (ota-meshi) [inline docs/rules/arrow-parens.md]:
```suggestion
    twich: (x => 2 * x),
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

**Review by ota-meshi — APPROVED** [review]
LGTM!

**Pull request merged**